vcfa/resource_vcfa_org_vdc.go

```diff
@@ -0,0 +1,263 @@
+			"org_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Parent Organization ID",
+			},
+			"region_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Parent Region ID",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: fmt.Sprintf("Description of the %s", labelVcfaOrgVdc),
+			},
+			"supervisor_ids": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: fmt.Sprintf("A set of Supervisor IDs that back this %s", labelVcfaOrgVdc),
+			},
+			"zone_resource_allocations": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        tmOrgVdcZoneResourceAllocation,
+				Description: "A set of Region Zones and their resource allocations",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: fmt.Sprintf("%s status", labelVcfaOrgVdc),
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: fmt.Sprintf("Name of the %s", labelVcfaOrgVdc),
+			},
+		},
+	}
+}
+
+var tmOrgVdcZoneResourceAllocation = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"region_zone_name": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: fmt.Sprintf("%s Name", labelVcfaRegionZone),
+		},
+		"region_zone_id": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: fmt.Sprintf("%s ID", labelVcfaRegionZone),
+		},
+		"memory_limit_mib": {
+			Type:             schema.TypeInt,
+			Required:         true,
+			Description:      "Memory limit in MiB",
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(0)),
+		},
+		"memory_reservation_mib": {
+			Type:             schema.TypeInt,
+			Required:         true,
+			Description:      "Memory reservation in MiB",
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(0)),
+		},
+		"cpu_limit_mhz": {
+			Type:             schema.TypeInt,
+			Required:         true,
+			Description:      "CPU limit in MHz",
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(0)),
+		},
+		"cpu_reservation_mhz": {
+			Type:             schema.TypeInt,
+			Required:         true,
+			Description:      "CPU reservation in MHz",
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(0)),
+		},
+	},
+}
+
+func resourceTmOrgVdcCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	c := crudConfig[*govcd.TmVdc, types.TmVdc]{
+		entityLabel:      labelVcfaOrgVdc,
+		getTypeFunc:      getTmVdcType,
+		stateStoreFunc:   setTmVdcData,
+		createFunc:       vcdClient.CreateTmVdc,
+		resourceReadFunc: resourceTmOrgVdcRead,
+	}
+	return createResource(ctx, d, meta, c)
+}
+
+func resourceTmOrgVdcUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Hitting API's NPE when trying to update `cpu_reservation_mhz`:

![Screenshot 2025-01-23 at 09 48 44](https://github.com/user-attachments/assets/f8f84be2-b410-4133-a911-a4b2adbdff69)

Reply:
Yep, there's a reported bug. Update does not work and is currently commented in test